packages-common/pluggable-widgets-tools/tests/index.spec.js

```diff
@@ -1,147 +1,76 @@
-        });
-
-        it(`Testing start:ts command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["start:ts"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "start:ts"`);
-        });
-
-        it(`Testing start:js command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["start:js"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "start:js"`);
-        });
-
-        it(`Testing build:ts command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["build:ts"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "build:ts"`);
-        });
+const workDir = join(tempdir(), `pwt_test_${Math.round(Math.random() * 10000)}`);
+let toolsPackagePath;
 
-        it(`Testing build:js command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["build:js"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "build:js"`);
-        });
-
-        it(`Testing start:ts command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["start:ts"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "start:ts"`);
-        });
+beforeAll(() => {
+    cd(join(__dirname, ".."));
+    const packResult = exec("npm pack --loglevel=error", { silent: true });
+    toolsPackagePath = join(__dirname, "..", packResult.stdout.trim());
 
-        it(`Testing dev:js command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["dev:js"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "dev:js"`);
-        });
-
-        it(`Testing dev:ts command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["dev:ts"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "dev:ts"`);
-        });
+    mkdir(workDir);
+    cd(workDir);
+});
+afterAll(() => {
+    rm("-rf", [workDir, toolsPackagePath]);
+});
 
-        it(`Testing release:js command`, function() {
-            const proc = spawnSync('pluggable-widgets-tools', ["release:js"]);
-            expect(proc.stdout).not.toBeNull();
-            expect(proc.stdout.toString()).toContain(`${expectedOutput} "release:js"`);
-        });
+describe.each([
+    ["web", "ts", "8.0"],
+    ["native", "js", "8.1"],
+    ["web", "js", "8.3"],
+    ["native", "ts", "8.6"],
+    ["web", "ts", "8.6"]
+])("For %s %s widget created with generator v%s", (platform, lang, version) => {
```

Comment:
As soon as we move generator, can we re-use it and force the real generation? Otherwise every time we change boilerplate templates we need to update generated projects.

Reply:
We never need to update them, since they are fixed projects created with old generators. But I think we should add a test (and here is a todo below) that uses a _current_ generator. After your comment I start to become hesitant of such test, because (correct me if i'm wrong) generator has similar tests => as soon as it is moved to monorepo there is no big value in having tests here that cover the same thing.